plugins/importexport/doaj/DOAJPlugin.inc.php

```diff
@@ -103,6 +144,314 @@ function exportJournal(&$journal, $outputFile = null) {
+			foreach($selectedArticles as $articleId) {
+				$article = $articleDao->getArticle($articleId);
+				$article->setData('doaj::registered', 1);
+				$articleDao->updateArticle($article);
+			}
+		}
+
+		// check for issues
+		$selectedIssues = $selectedObjects[DOAJ_EXPORT_ISSUES];
+		if (is_array($selectedIssues) && !empty($selectedIssues)) {
+			$issueDao = DAORegistry::getDAO('IssueDAO');
+
+			foreach($selectedIssues as $issueId) {
+				$articles = $this->_retrieveArticlesByIssueId($issueId);
+				foreach($articles as $article) {
+					$article->setData('doaj::registered', 1);
+					$articleDao->updateArticle($article);
+				}
+			}
+		}
+
+		// show message & redirect
+		$this->_sendNotification(
+			$request,
+			'plugins.importexport.doaj.markRegistered.success',
+			NOTIFICATION_TYPE_SUCCESS
+		);
+
+		$request->redirect(null, null, null, array('plugin', $this->getName(), $action));
+	}
+
+	/**
+	 * Display a list of issues for export.
+	 * @param $templateMgr TemplateManager
+	 * @param $journal Journal
+	 */
+	function _displayIssueList($templateMgr, $journal) {
+		$this->setBreadcrumbs(array(), true);
+
+		// Retrieve all published issues.
+		AppLocale::requireComponents(array(LOCALE_COMPONENT_OJS_EDITOR));
+		$issueDao = DAORegistry::getDAO('IssueDAO'); /* @var $issueDao IssueDAO */
+		$issueIterator = $issueDao->getPublishedIssues($journal->getId(), Handler::getRangeInfo('issues'));
+
+		// check whether all articles of an issue are doaj::registered or not
+		$issues = array();
+		while ($issue = $issueIterator->next()) {
+			$issueId = $issue->getData('id');
+			$articles = $this->_retrieveArticlesByIssueId($issueId);
+			$allArticlesRegistered = true;
+
+			foreach($articles as $article) {
+				if (!$article->getData('doaj::registered')) {
+					$allArticlesRegistered = false;
+					break;
+				}
+			}
+
+			$issue->setData('doaj::registered', $allArticlesRegistered);
+			$issues[] = $issue;
+			unset($issue);
+		}
+		unset($issueIterator);
+
+		// Instantiate issue iterator.
+		import('lib.pkp.classes.core.ArrayItemIterator');
+		$rangeInfo = Handler::getRangeInfo('articles');
+		$iterator = new ArrayItemIterator($issues, $rangeInfo->getPage(), $rangeInfo->getCount());
+
+		// Prepare and display the issue template.
+		$templateMgr->assign_by_ref('issues', $iterator);
+		$templateMgr->display($this->getTemplatePath() . 'issues.tpl');
+	}
+
+	/**
+	 * Display a list of articles for export.
+	 * @param $templateMgr TemplateManager
+	 * @param $journal Journal
+	 * @param $unregistered boolean
+	 */
+	function _displayArticleList($templateMgr, $journal, $unregistered = false) {
+		$this->setBreadcrumbs(array(), true);
+
+		// Retrieve all published articles.
+		$articles = $this->_getAllPublishedArticles($journal);
+
+		// Retrieve array elements without index "doaj::registered"
+		if ($unregistered == true) {
+			$articles = array_filter($articles, function($article) { return !$article['article']->getData('doaj::registered'); });
```

Comment:
Check this syntax against our PHP version requirements -- consider using create_function instead.